pecos/core/ann/hnsw.hpp

```diff
@@ -870,5 +1009,343 @@ namespace ann {
     };
 
 
-} // end of namespace ann
-} // end of namespace pecos
+    template<typename dist_t>
+    struct HNSWProductQuantizer4Bits {
+        typedef FeatVecDenseL2Simd<float> feat_vec_t;
```

Comment:
we currently only support `L2`, need to add some warning.

Reply:
Currently we don't have parameter to let user input distance metric. Maybe it's easier to write it in the README.md in the example folder ?